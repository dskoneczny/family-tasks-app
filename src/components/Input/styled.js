import styled from 'styled-components'
import { Colors } from '../../lib/colors'

export const InputStyled = styled.TextInput`
  background: ${Colors.background};
  padding: 16px 12px;
  flex: 1;
`