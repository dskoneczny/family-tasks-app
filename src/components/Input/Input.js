import React from 'react'
import { InputStyled } from './styled'

export const Input = (props) => {
  const { inputProps } = props
  return (
    <InputStyled {...inputProps} />
  )
}