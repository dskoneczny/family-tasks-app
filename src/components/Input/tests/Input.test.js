import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import { Input } from '../Input'

test('input value change', async () => {
  const onPressMock = jest.fn();
  const expectedText = "new content"
  const { getByPlaceholderText, toJSON } = render(
    <Input
      inputProps={{
        value: "",
        placeholder: 'Placeholder',
        onChangeText: onPressMock,
      }}
    />)

  fireEvent.changeText(getByPlaceholderText('Placeholder'), expectedText);
  expect(onPressMock).toHaveBeenCalledWith(expectedText);

  expect(toJSON()).toMatchSnapshot()
})

test('default input value', async () => {
  const onPressMock = jest.fn();

  const { toJSON } = render(
    <Input
      inputProps={{
        value: "Default Value",
        placeholder: 'Placeholder',
        onChangeText: onPressMock,
      }}
    />)


  expect(toJSON()).toMatchSnapshot()
})