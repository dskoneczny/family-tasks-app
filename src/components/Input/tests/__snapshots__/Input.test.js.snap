// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default input value 1`] = `
<TextInput
  onChangeText={[MockFunction]}
  placeholder="Placeholder"
  style={
    Array [
      Object {
        "backgroundColor": "#F2F5FC",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 16,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 16,
      },
    ]
  }
  value="Default Value"
/>
`;

exports[`input value change 1`] = `
<TextInput
  onChangeText={
    [MockFunction] {
      "calls": Array [
        Array [
          "new content",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="Placeholder"
  style={
    Array [
      Object {
        "backgroundColor": "#F2F5FC",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 16,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 16,
      },
    ]
  }
  value=""
/>
`;
